protocol/x/clob/keeper/process_operations.go

```diff
@@ -248,6 +214,218 @@ func (k Keeper) PersistMatchToState(
 	return nil
 }
 
+// PersistOrderRemovalToState takes in an OrderRemoval, statefully validates it according to
+// RemovalReason, and writes the removal to state.
+func (k Keeper) PersistOrderRemovalToState(
+	ctx sdk.Context,
+	orderRemoval *types.OrderRemoval,
+) error {
+	orderIdToRemove := orderRemoval.GetOrderId()
+	orderIdToRemove.MustBeStatefulOrder()
+
+	// Order removals are always for stateful orders that must exist.
+	orderToRemove, err := k.FetchOrderFromOrderId(ctx, orderIdToRemove, make(map[types.OrderId]types.Order, 0))
```

Comment:
`FetchOrderFromOrderId`  only fetches orders from long term store and *triggered* store. it does not fetch untriggered conditional orders. 

I think removals for untriggered conditional orders is still valid, for example due to collateralization reasons, so using `FetchOrderFromOrderId` is technically not always correct.

Reply:
I thought we have removal tests for untriggered conditional orders, but i could be wrong. if we don't can we add them

Reply:
How could we have Order Removals for untriggered conditional orders? Order Removals are triggered by the matching loop, but untriggered orders are untriggered and have never been placed. This means they can't be removed as a taker because that means they are triggered. They can't be removed as makers because they aren't on the book.

Reply:
yeah previously I thought it can fail collat check. but I guess in that case, we should not even save that order to state in the first place. 

@jonfung-dydx can you confirm this is true?

Reply:
I believe that when we place the stateful conditional order in DeliverTx, we run a collat check [here](https://github.com/dydxprotocol/v4-chain/blob/main/protocol/x/clob/keeper/orders.go#L208-L223). So I think it's ok if we don't check untriggered store. If we place it and it is collateralized, then it goes undercollateralized, when the order is triggered, we will move from untriggered state to triggered state in EndBlocker. Then, when we attempt to place the order via `PlaceConditionalOrdersTriggeredInLastBlock` in PrepareCheckState, we will always be generating removal reasons off of triggered conditional orders.

Reply:
^ yep that sounds exactly right to me